odoo/addons/base/models/ir_cron.py

```diff
@@ -290,91 +300,182 @@ def _acquire_one_job(cls, cr, job_ids):
+                progress = cron._get_progress()
+                job_cr.commit()
+
+                try:
+                    cron._callback(job['cron_name'], job['ir_actions_server_id'])
+                except Exception:
+                    # If some progress has been committed, we do not consider it a failure
+                    status = 'PARTIALLY_DONE' if progress.done and progress.remaining else 'FAILED'
+                else:
+                    if not progress.done or not progress.remaining:
+                        status = 'FULLY_DONE'
+                    else:
+                        status = 'PARTIALLY_DONE'
+                finally:
+                    progress.timed_out_counter = 0
+                    timed_out_counter = 0
+                    job_cr.commit()
+                _logger.info('Job `%s` (%s): Processed %s records, %s records remaining.',
+                             job['cron_name'], job['id'], progress.done, progress.remaining)
+                if status in ['FULLY_DONE', 'FAILED']:
+                    break
+
+        return status
+
+    def _update_failure_count(self, cr, job, status):
+        """ Update cron `failure_count` and `first_failure_date`.
+        Count is increased on failure and both fields are reset on success.
+        Cron that fail MIN_FAILURE_COUNT_BEFORE_DEACTIVATION times in a row over a period at least greater
+        than MIN_DELTA_BEFORE_DEACTIVATION are deactivated.
+        """
+        now = fields.Datetime.context_timestamp(self, datetime.utcnow())
+
+        if status == 'FAILED':
+            failure_count = job['failure_count'] + 1
+            first_failure_date = job['first_failure_date'] or now
+            active = True
+            if (
+                failure_count >= MIN_FAILURE_COUNT_BEFORE_DEACTIVATION
+                and fields.Datetime.context_timestamp(self, first_failure_date) + MIN_DELTA_BEFORE_DEACTIVATION < now
+            ):
+                failure_count = 0
+                first_failure_date = None
+                active = False
+                self._notify_admin_deactivation(job['cron_name'], job['id'], job['ir_actions_server_id'])
+        else:
+            failure_count = 0
+            first_failure_date = None
+            active = True
 
-            # The actual cron execution
-            for call in range(effective_call_count):
-                ir_cron._callback(job['cron_name'], job['ir_actions_server_id'], job['id'])
+        cr.execute("""
+            UPDATE ir_cron
+            SET failure_count = %s,
+                first_failure_date = %s,
+                active = %s
+            WHERE id = %s
+        """, [
+            failure_count,
+            first_failure_date,
+            active,
+            job['id'],
+        ])
 
-        # Update the cron with the information computed above
-        cron_cr.execute("""
+    def _mark_as_done(self, cr, job):
```

Comment:
`mark_as_done` was very clear to me, suggestion:
```suggestion
    def _reschedule(self, cr, job):
```

Reply:
But it's the opposite of the rescheduling, it's marking it as : do not reschedule (or it's been `done` :eyes: )

Reply:
Julien meant "reschedule it later, according to its periodicity".